mobile/lib/services/rest_api.dart

```diff
@@ -153,35 +182,53 @@ class AirqoApiClient {
       );
     }
 
-    return forecasts;
+    return forecasts.removeInvalidData();
   }
 
-  Future<EmailAuthModel?> requestEmailVerificationCode(
-    String emailAddress,
-    bool reAuthenticate,
-  ) async {
+  Future<EmailAuthModel?> sendEmailVerificationCode(String emailAddress) async {
     try {
-      Map<String, String> headers = HashMap()
-        ..putIfAbsent(
-          'Content-Type',
-          () => 'application/json',
-        );
+      Map<String, String> headers = Map.from(postHeaders);
+      headers["service"] = ApiService.auth.serviceName;
+
+      final response = await client.post(
+        Uri.parse(
+          "${AirQoUrls.emailVerification}?TOKEN=${Config.airqoApiV2Token}",
+        ),
+        headers: headers,
+        body: jsonEncode({'email': emailAddress}),
+      );
+
+      return EmailAuthModel.fromJson(
+        json.decode(response.body) as Map<String, dynamic>,
+      );
+    } catch (exception, stackTrace) {
+      await logException(
+        exception,
+        stackTrace,
+      );
+    }
 
-      final body = {
-        'email': emailAddress,
-      };
+    return null;
+  }
 
-      final uri = reAuthenticate
-          ? AirQoUrls.requestEmailReAuthentication
-          : AirQoUrls.requestEmailVerification;
+  Future<EmailAuthModel?> sendEmailReAuthenticationCode(
+    String emailAddress,
+  ) async {
+    try {
+      Map<String, String> headers = Map.from(postHeaders);
+      headers["service"] = ApiService.auth.serviceName;
```

Comment:
Adding service names to headers @Baalmart

Reply:
thanks @NoahNsimbe